shared/checks/oval/installed_env_has_grub2_package.xml

```diff
@@ -31,7 +31,7 @@
     <linux:object object_ref="obj_env_has_grub2_installed" />
   </linux:rpminfo_test>
   <linux:rpminfo_object id="obj_env_has_grub2_installed" version="1">
-    <linux:name>grub2-common</linux:name>
+    <linux:name>{{% if product == "sle12" %}}grub2{{% else %}}grub2-common{{% endif %}}</linux:name>
```

Comment:
I recommend you add `"grub2": "grub2",` in `sle12/product.yml`, similar to what was done with `login_defs`/ `shadow`.
https://github.com/ComplianceAsCode/content/blob/2ddb4860a56b1f18765fdbfeccbb85b3a6a1917c/sle12/product.yml#L27

Reply:
Isn't that `product.yml` definition used only for bash and ansible remediation? to detect the correct package before applying the remediation. I'm afraid that there is no other option than this jinja `if`. Someone already managed to include a similar piece of code: https://github.com/ComplianceAsCode/content/blob/5f5f5f011be2b59fa86c960f61dcad1b0af96417/shared/checks/oval/installed_env_has_login_defs.xml#L28

Reply:
Thanks for the suggestion. Updating sle12/product.yml seem to have worked as well. I'll make the change.

Reply:
> Isn't that `product.yml` definition used only for bash and ansible remediation? to detect the correct package before applying the remediation. I'm afraid that there is no other option than this jinja `if`. Someone has already managed to include a similar piece of code:
> 
> https://github.com/ComplianceAsCode/content/blob/5f5f5f011be2b59fa86c960f61dcad1b0af96417/shared/checks/oval/installed_env_has_login_defs.xml#L28

I retested the change and updating product.yml seem to have worked as well.

Reply:
> Isn't that `product.yml` definition used only for bash and ansible remediation?

Yes, used for bash and ansible remediation. 

But the change in ` shared/checks/oval/installed_env_has_grub2_package.xml ` is also required, so that the OVAL checks for the same package.

I think both changes are actually needed, sorry for not being clear.

Reply:
I re-ran the OVAL and it seem to have correctly detected the grub2 package with just the mapping in product.yml. Though I did not test Ansible or Bash as the rule has no auto-remediation. Please let me know what needs to be validated if the change is needed in both places.

Reply:
Hmm, try checking and remediating `grub2_audit_argument`.

Reply:
Looks like we need the changes in `shared/checks/oval/installed_env_has_grub2_package.xml` as well. I was testing with the wrong local branch earlier. My apologies.

Reply:
No worries.